homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -114,6 +123,97 @@ def get_platforms(config_entry):
     return []
 
 
+def _async_update_data_default(hass, device):
+    async def update():
+        """Fetch data from the device using async_add_executor_job."""
+        try:
+            async with async_timeout.timeout(10):
+                state = await hass.async_add_executor_job(device.status)
+                _LOGGER.debug("Got new state: %s", state)
+                return state
+
+        except DeviceException as ex:
+            raise UpdateFailed(ex) from ex
+
+    return update
+
+
+@dataclass(frozen=True)
+class VacuumCoordinatorData:
+    """A class that holds the vacuum data retrieved by the coordinator."""
+
+    status: VacuumStatus
+    dnd_status: DNDStatus
+    last_clean_details: CleaningDetails
+    consumable_status: ConsumableStatus
+    clean_history_status: CleaningSummary
+    timers: List[Timer]
+    fan_speeds: Dict[str, int]
+    fan_speeds_reverse: Dict[int, str]
+
+
+@dataclass(init=False, frozen=True)
+class VacuumCoordinatorDataAttributes:
+    """
+    A class that holds attribute names for VacuumCoordinatorData.
+
+    These attributes can be used in methods like `getattr` when a generic solutions is needed.
+    See homeassistant.components.xiaomi_miio.device.XiaomiCoordinatedMiioEntity._extract_value_from_attribute for
```

Comment:
Please break longs strings around 88 characters. Docstring lines should preferably not be longer than 72 characters according to PEP8.

Reply:
Linter did not complain tho :(

Reply:
Fixed my editor settings 👍🏾